File: app/src/main/java/vgame/ir/data/model/Teacher.java
package vgame.ir.data.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.Objects;

public class Teacher {

    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("description")
    @Expose
    private String description;
    @SerializedName("user_id")
    @Expose
    private Integer userId;
    @SerializedName("courses_count")
    @Expose
    private Integer coursesCount;
    @SerializedName("user")
    @Expose
    private User user;

    @SerializedName("avatar_url")
    @Expose
    private String avatar_url;

    public String getAvatar_url() {
        return avatar_url;
    }

    public void setAvatar_url(String avatar_url) {
        this.avatar_url = avatar_url;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Teacher teacher = (Teacher) o;
        return Objects.equals(id, teacher.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public Integer getCoursesCount() {
        return coursesCount;
    }

    public void setCoursesCount(Integer coursesCount) {
        this.coursesCount = coursesCount;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

}

File: app/src/main/java/vgame/ir/data/model/Teacher.kt
package vgame.ir.data.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.util.Objects

class Teacher {

    @SerializedName("id")
    @Expose
    var id: Int? = null
    @SerializedName("description")
    @Expose
    var description: String? = null
    @SerializedName("user_id")
    @Expose
    var userId: Int? = null
    @SerializedName("courses_count")
    @Expose
    var coursesCount: Int? = null
    @SerializedName("user")
    @Expose
    var user: User? = null

    @SerializedName("avatar_url")
    @Expose
    var avatar_url: String? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val teacher = o as Teacher?
        return id == teacher!!.id
    }

    override fun hashCode(): Int {
        return Objects.hash(id)
    }

}
